### ISSUE TITLE
Snap to grid compliant with page template grid size

### ISSUE THREAD

**Issue by begbaj** [xournalpp/xournalpp#4688]
**Is your feature request related to a problem? Please describe.**
I felt the need to have at least one more graph page template, since the default one is sticked to 5mm in grid size, while what I want is 4mm. So I decided to add a custom page template, but the problem with this is that "snap to grid" feature will incorrectly snap to a grid of 5mm (14.17 pixels) .
This tool has a setting in the Settings.h header that sticks the grid size of the "snap to grid" tool to 14.17 (5mm), making other page tamplates very uncomfortable to deal with.

**Describe the solution you'd like**
"snap to grid" tool should get the "grid size" from the currently focused page, not from a default universal static value.

**Comment by rolandlo** [comment]
You're right that this should be improved. If you are working with 4mm grid size all the time, you can set the `grid size` to `0.8` in the "Drawing area" tab in menu `Edit > Preferences` currently.

**Comment by begbaj** [comment]
> You're right that this should be improved. If you are working with 4mm grid size all the time, you can set the `grid size` to `0.8` in the "Drawing area" tab in menu `Edit > Preferences` currently.

Fantastic, this is a perfect workaround. Anyway, I think that this is an important improvement to userfriendlyness, so I certainly will look on how this can be done when I find the time to do so 😄. Thank you for the response
